File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.py
class Solution:
    def maxVowels(self, s: str, k: int) -> int:
        n = len(s)
        if n < k:
            return 0
        vowels = set(['a', 'e', 'i', 'o', 'u'])
        i = 0
        j = k-1
        res = sum(s[l] in vowels for l in range(k))
        current = res
        while j < (n - 1):
            j += 1
            i += 1
            if s[i-1] in vowels:
                current -= 1
            if s[j] in vowels:
                current += 1
            res = max(res, current)
        return res
            

File: 1456-maximum-number-of-vowels-in-a-substring-of-given-length/1456-maximum-number-of-vowels-in-a-substring-of-given-length.rs
use std::cmp;
use std::collections::HashSet;

impl Solution {
    pub fn max_vowels(s: String, k: i32) -> i32 {
        let n: i32 = s.len() as i32;
        if (n < k) {
            return 0;
        }
        let s_vec: Vec<char> = s.chars().collect();
        let vowels: HashSet<char> = HashSet::from(['a', 'e', 'i', 'o', 'u']);
        let mut i: usize = 0;
        let mut j: usize = (k as usize) - 1;
        let mut res: i32 = 0;
        for l in 0..k as usize {
            if (vowels.contains(&(s_vec[l]))) {
                res += 1;
            }
        }
        let mut current: i32 = res;
        while (j < ((n as usize) - 1)) {
            j += 1;
            i += 1;
            if (vowels.contains(&(s_vec[i-1]))) {
                current -= 1;
            }
            if (vowels.contains(&(s_vec[j]))) {
                current += 1;
            }
            res = cmp::max(res, current);
        }
        return res;
    }
}
